lib/improver/database.py

```diff
@@ -0,0 +1,248 @@
+        Ensure that the dimensions and coordinates are shared between cubes.
+
+        """
+        cubelist = self.cubelist
+        some_cube = self.cubelist[0]
+
+        for cube in cubelist:
+            for coord in cube.dim_coords:
+                assert coord.is_compatible(some_cube.coord(coord.name()))
+
+    def to_dataframe(self):
+        """
+        Turn the input cubes into a 2-dimensional DataFrame object
+
+        """
+
+        for cube in self.cubelist:
+            for c in cube.slices_over(self.column_dims[0]):
+                df = DataFrame(c.data,
+                               index=c.coord(self.primary_dim).points,
+                               columns=['vals'])
+
+                if self.pivot_dim:
+                    # Reshape data based on column values
+                    coords = c.coord(self.pivot_dim).points
+                    col_names = map(self.pivot_map, coords)
+                    df.insert(1, self.pivot_dim, col_names)
+                    df = df.pivot(columns=self.pivot_dim, values='vals')
+
+                if self.primary_map:
+                    # Switch the index out for a map if specified
+                    for mapping, function in zip(self.primary_map,
+                                                 self.primary_func):
+                        df.insert(0, mapping, map(function, df.index))
+
+                    # Takes significant time if a multi-index
+                    df.set_index(self.primary_map, inplace=True)
+
+                for dim, col in itertools.izip_longest(self.column_dims,
+                                                       self.column_maps):
+                    if dim in df.columns:
+                        continue
+                    
+                    if dim in [coord.standard_name or coord.long_name 
+                               for coord in c.dim_coords + c.aux_coords]:
+                        coord = c.coord(dim)
+                        column_name = col or dim
+                        if len(coord.points) == 1:
+                            column_data = coord.points[0]
+                        else:
+                            column_data = coord.points
+                    
+                    # Check to see if provided dim is a method of the cube
+                    elif hasattr(c, dim):
+                        attr = getattr(c, dim)
+                        if callable(attr):
+                            column_name = col
+                            column_data = attr()
+                            
+                    else:
+                        column_name = col
+                        column_data = dim
+                   
+                    as_index = True if dim != self.pivot_dim else False
+                    self.insert_column(df, column_name, column_data, as_index)
+                        
+                try:
+                    self.df = self.df.combine_first(df)
+                except AttributeError:
```

Comment:
This isn't where the error is being raised, and this hasn't fixed the problem. The error gets raised when you have saved the first iteration and then try to save more data to the same file.